### DISCUSSION TITLE
Does this bypass US sanctions?

### DISCUSSION THREAD

**Discussion by ehsan-g** [OutlineFoundation/outline-sdk#97] [Q&A]
I came across an article suggesting that this method could potentially assist Iranians in circumventing censorship imposed in Iran. However, as a developer residing in Iran, I can't help but feel the impact of discrimination caused by the US technology sanctions. Numerous third-party libraries yield a 403 error, and it's rather ironic that US-based companies, particularly those operating in the blockchain realm, prominently feature the term "decentralized" on their websites, all while they have imposed IP address restrictions on Iranians. WalletConnect serves as a prime example. My inquiry revolves around whether this approach can also aid in bypassing US sanctions?

**Comment by MartinHennrich** [comment]
I suppose Outline should be used in Iran but I fear that the government will punish you in that case.

**Comment by amircybersec** [comment]
@ehsan-g Unfortunately large tech companies still block Iran IP addresses from accessing some of their services (especially enterprise, cloud services, etc) in order to comply with US sanctions. Some of these policies are unfortunately hurting the citizens of Iran who depend on many of these infrastructures to bypass censorship. Some serious policy making in US needs to be done there to relax some of these restrictions. 

When it comes to hosting an Outline VPN server, there are many VPS service providers that are technically accessible from Iran. Outline technology itself is open source and is not tied to a specific service provider. 

@MartinHennrich Outline clients and other VPN services are widely used in Iran albeit the inherent risk. Almost every Iranian citizen online these days uses some kind of VPN service to bypass censorship. The sheer scale and number of users for these services makes it almost impossible to the government to criminalize/prosecute.

**Comment by ehsan-g** [comment]
@amircybersec I was hoping to use it inside my PWA/web App, but apparently only available for Android and iOS.

**Comment by amircybersec** [comment]
@ehsan-g I think it would be possible to compile the golang application into web assembly (.wasm) to run it inside the browser. I have not tested or done this before but I think it would be possible. Here's some pointer: https://www.sitepen.com/blog/compiling-go-to-webassembly

**Reply by fortuna** [reply]
Webassembly doesn't have access to the transport layer, only fetch. There's not much that can be done on the browser.

You can, however, run a local proxy and configure the browser to use it.
